jbmc/unit/java_bytecode/java_bytecode_language/language.cpp

```diff
@@ -0,0 +1,47 @@
+/*******************************************************************\
+
+Module: Unit tests for java_bytecode_language.
+
+Author: Diffblue Limited.
+
+\*******************************************************************/
+
+#include <testing-utils/use_catch.h>
+
+#include <util/symbol_table.h>
+
+#include <java-testing-utils/load_java_class.h>
+#include <java-testing-utils/require_type.h>
+
+SCENARIO(
+  "java_bytecode_language_opaque_field",
+  "[core][java_bytecode][java_bytecode_language]")
+{
+  GIVEN("A class that accesses opaque fields")
+  {
+    const symbol_tablet symbol_table = load_java_class(
+      "ClassUsingOpaqueField", "./java_bytecode/java_bytecode_language");
+    std::string opaque_class_prefix = "java::OpaqueClass";
+
+    WHEN("When parsing opaque class with fields")
+    {
+      THEN("Static field field1 is marked as final")
+      {
+        const symbolt &opaque_field_symbol =
+          symbol_table.lookup_ref(opaque_class_prefix + ".field1");
+        REQUIRE(opaque_field_symbol.type.get_bool(ID_C_constant));
```

Comment:
:pick: @thomasspriggs was going to look at a nice way of accessing properties of static fields (since them being comments on the type is a bit unstable) perhaps in a follow up PR you can do whatever he ends up doing.

Reply:
Good point, Tom passed me his PRs so I will follow up on that.